lib/pymedphys/_dicom/connect/listen.py

```diff
@@ -0,0 +1,207 @@
+# Copyright (C) 2020 University of New South Wales & Ingham Institute
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import signal
+import sys
+import tempfile
+from pathlib import Path
+
+from pydicom.dataset import Dataset, FileDataset
+from pynetdicom import (
+    AE,
+    PYNETDICOM_IMPLEMENTATION_UID,
+    PYNETDICOM_IMPLEMENTATION_VERSION,
+    StoragePresentationContexts,
+    evt,
+)
+from pynetdicom.sop_class import (  # pylint: disable=no-name-in-module
+    VerificationSOPClass,
+)
+
+from pymedphys._dicom.connect.base import DicomConnectBase
+
+
+class DicomListener(DicomConnectBase):
+    def __init__(
+        self, storage_directory=tempfile.mkdtemp(), on_released_callback=None, **kwargs
+    ):
+        super(DicomListener, self).__init__(**kwargs)
+
+        # The directory where incoming data will be stored
+        self.storage_directory = storage_directory
```

Comment:
this could be reduced to 
self.storage_directory = Path(storage_directory)
if storage_directory is already a Path, it still comes out as a Path